detox/e2e/test/messaging/message_posting.e2e.js

```diff
@@ -26,4 +29,23 @@ describe('Messaging', () => {
 
         await expect(element(by.text(text))).toExist();
     });
+
+    it('MM-T3187 Unread channels sort at top', async () => {
+        // # Open channel drawer (with at least one unread channel)
+        await element(by.id('channel_drawer_button')).tap();
+
+        // # Verify unread channel(s) display at top of channel list (with mentions first, if any), in alphabetical order, with title "Unreads"
+        await expect(element(by.text('UNREADS'))).toBeVisible();
+
+        // # Tap an unread channel to view it
+        await element(by.text(newChannel.display_name)).tap();
+
+        // # Open channel drawer again
+        await element(by.id('channel_drawer_button')).tap();
+
+        // * Channel you just read is no longer listed in Unreads
+        await expect(element(by.text('PUBLIC CHANNELS'))).toBeVisible();
+        await expect(element(by.text('UNREADS'))).not.toBeVisible();
+        await expect(element(by.text(newChannel.display_name)).atIndex(1)).toBeVisible();
```

Comment:
There's `newChannel.display_name` on the navbar and the channel list. I've tried `withAncestor(by.type('SectionList'))` but no avail.  Using `atIndex(1)` is not good. If there's an XPath-like syntax to specify the channel name from the channe list, please let me know.

Reply:
Hi @clarmso, I think `by.type()` assumes the native class type from iOS or Android (https://github.com/wix/Detox/blob/master/docs/APIRef.Matchers.md#bytypeclassname). Maybe a workaround is to add `id` to `SectionList` and use `withAncestor(by.id(...))` instead. I haven't tried it yet but maybe worth to give a shot.

Reply:
Adding testID to `SectionList` does not work for me, but I could do so with `ChannelList`.  Still, I can't get `withDescendants` and `withAncestor` working. I've tried the following:

`await expect(element(by.id('channels_list').withDescendant(by.id('channel_item_display_name')))).toExist();`
`await expect(element(by.text(newChannel.display_name).withAncestor(by.id('channels_list')))).toExist();`

Turning on verbose (on --loglevel) does not show info.

In addition, I've tried the following. `element(by.id('channel_item_display_name')).atIndex(0)`) points to "Off-Topic" despite react devtools shows that index 0 points to "Channel <random>" 

`await expect(element(by.id('channel_item_display_name')).atIndex(0)).toHaveText(newChannel.display_name);`

Reply:
Hi @clarmso I believe you added `testID='channels_list'` to the title of the list and not to the enclosing container of the list (`channels_list.js`). I added it to the container as shown below and both assertions passed for me locally,

![Screen Shot 2020-09-28 at 11 31 05 AM](https://user-images.githubusercontent.com/487991/94472092-146b9000-017f-11eb-96ca-7202fab542ba.png)

```
await expect(element(by.id('channels_list').withDescendant(by.id('channel_item_display_name')))).toExist();
await expect(element(by.text(newChannel.display_name).withAncestor(by.id('channels_list')))).toExist();
```